app/models/role.rb

```diff
@@ -33,8 +29,20 @@ def can_set_role?
     admin? ? true : permision? && permission.can_set_role
   end
 
-  def can_edit?
-    editor? ? true : permision? && permission.can_set_role
+  def can_read_entities?
+    editor? ? true : permision? && permission.can_read_entities
+  end
+
+  def can_edit_entities?
+    editor? ? true : permision? && permission.can_edit_entities
+  end
+
+  def can_read_movie_and_person?
+    predefined_user? ? true : permision? && permission.can_read_movie_and_person
+  end
+
+  def can_edit_movie_and_person?
+    editor? ? true : permision? && permission.can_edit_movie_and_person
```

Comment:
solid and kiss. If you don t see problem please write me

Reply:
You don t break first solid  principe. IT about classes not methods. BUT methods MUST have one responsibility TOO. I think you DON T follow KISS. You method too complicated and NOT universal. PLEASE can_edit_movie? can_edit_smt? BUT avoid AND

Reply:
ok thanks, i'll fix it now